File: examples/direct/TextConsumer.java
import org.apache.thrift.TException;
import org.apache.thrift.transport.TTransport;
import org.apache.thrift.transport.TSocket;
import org.apache.thrift.transport.TTransportException;
import org.apache.thrift.protocol.TBinaryProtocol;
import org.apache.thrift.protocol.TProtocol;
import uk.co.fredemmott.jp.*;

public class TextConsumer
{
	public static void main(String[] args)
	{
		try
		{
			TTransport transport = new TSocket("localhost", 9090);
			TProtocol protocol = new TBinaryProtocol(transport);
			JobPool.Client client = new JobPool.Client(protocol);

			transport.open();

			while(true)
			{
				Job job = client.acquire("text");
				System.out.print("I'm consuming a ");
				System.out.println(new String(job.message, "UTF-8"));
				client.purge("text", job.id);
				System.out.println("Consumed.");
			}
		}
		catch (java.io.UnsupportedEncodingException e)
		{
			System.out.println("UTF-8 encoding not supported, couldn't process item.");
		}
		catch (NoSuchPool e)
		{
			System.out.println("Pool 'text' does not exist, couldn't acquire item.");
		}
		catch (EmptyPool e)
		{
			System.out.println("Pool is empty :(");
		}
		catch (TException e)
		{
			e.printStackTrace();
		}
	}
}

File: examples/direct/TextConsumer.cpp
#include "Jp.h"

#include <protocol/TBinaryProtocol.h>
#include <transport/TSocket.h>
#include <transport/TTransportUtils.h>

#include <iostream>

using namespace apache::thrift;
using namespace apache::thrift::protocol;
using namespace apache::thrift::transport;

using boost::shared_ptr;

using namespace std;

int main(int argc, char** argv)
{
	shared_ptr<TTransport> socket(new TSocket("localhost", 9090));
	shared_ptr<TTransport> transport(new TBufferedTransport(socket));
	shared_ptr<TProtocol> protocol(new TBinaryProtocol(transport));
	JpClient client(protocol);
	transport->open();

	while(true)
	{
		try
		{
			Job job;
			client.acquire(job, "text");
			cout << "I'm consuming a " << job.message << "." << endl;
			client.purge("text", job.id);
			cout << "Consumed a " << job.message << "." << endl;
		}
		catch ( EmptyPool e)
		{
			cout << "Pool is empty." << endl;
			break;
		}
	}
	return 0;
}
